packages/vaex-core/vaex/__init__.py

```diff
@@ -474,13 +475,99 @@ def from_json(path_or_buffer, orient=None, precise_float=False, lines=False, cop
                        copy_index=copy_index)
 
 
-def from_csv(filename_or_buffer, copy_index=True, **kwargs):
-    """Shortcut to read a csv file using pandas and convert to a DataFrame directly.
-
-    :rtype: DataFrame
+def from_csv(filename_or_buffer, copy_index=True, chunk_size=None, convert=False, **kwargs):
     """
+    Read a CSV file as a DataFrame, and optionally convert to an hdf5 file.
+
+    :param str or file filename_or_buffer: CSV file path or file-like
+    :param bool copy_index: copy index when source is read via Pandas
+    :param int chunk_size: if the CSV file is too big to fit in the memory this parameter can be used to read
+        CSV file in chunks. For example:
+
+        >>> import vaex
+        >>> for i, df in enumerate(vaex.from_csv('taxi.csv', chunk_size=100_000)):
+        >>>     df = df[df.passenger_count < 6]
+        >>>     df.export_hdf5(f'taxi_{i:02}.hdf5')
+
+    :param bool or str convert: convert files to an hdf5 file for optimization, can also be a path. The CSV
+        file will be read in chunks: either using the provided chunk_size argument, or a default size. Each chunk will
+        be saved as a separate hdf5 file, then all of them will be combined into one hdf5 file. So for a big CSV file
+        you will need at least double of extra space on the disk. Default chunk_size for converting is 5 million rows,
+        which corresponds to around 1Gb memory on an example of NYC Taxi dataset.
+    :param kwargs: extra keyword arguments, currently passed to Pandas read_csv function, but the implementation might
+        change in future versions.
+    :returns: DataFrame
+    """
+    if not convert:
+        return _from_csv_read(filename_or_buffer=filename_or_buffer, copy_index=copy_index,
+                              chunk_size=chunk_size, **kwargs)
+    else:
+        if chunk_size is None:
+            # make it memory efficient by default
+            chunk_size = 5_000_000
+        return _from_csv_convert_and_read(filename_or_buffer=filename_or_buffer, copy_index=copy_index,
+                                          maybe_convert_path=convert, chunk_size=chunk_size, **kwargs)
+
+
+def _from_csv_read(filename_or_buffer, copy_index, chunk_size, **kwargs):
+    import pandas as pd
+    if not chunk_size:
+        full_df = pd.read_csv(filename_or_buffer, **kwargs)
+        return from_pandas(full_df, copy_index=copy_index)
+    else:
+        def iterator():
+            chunk_iterator = pd.read_csv(filename_or_buffer, chunksize=chunk_size, **kwargs)
+            for chunk_df in chunk_iterator:
+                yield from_pandas(chunk_df, copy_index=copy_index)
+        return iterator()
+
+
+def _from_csv_convert_and_read(filename_or_buffer, copy_index, maybe_convert_path, chunk_size, **kwargs):
+    # figure out the CSV file path
+    if isinstance(filename_or_buffer, six.string_types):
```

Comment:
since we now use Python3 only, we don't have to do this anymore. I don't want to change the current code aggressively to pollute the git history, that's why you still see Python2 compatible code 
```suggestion
    if isinstance(filename_or_buffer, str):
```

Reply:
Oh nice, looks better! Will use `str` then.